recipes/libssh/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "crypto_backend": "openssl",
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/[>=1.2.11 <2]")
+        if self.options.crypto_backend =="openssl":
+            self.requires("openssl/[>=1.1 <4]")
+        elif self.options.crypto_backend == "gcrypt":
+            self.requires("libgcrypt/1.8.4")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CLIENT_TESTING"] = False
+        tc.variables["SERVER_TESTING"] = False
+        tc.variables["WITH_EXAMPLES"] = False
+        tc.variables["WITH_GCRYPT"] = self.options.crypto_backend == "gcrypt"
+        tc.variables["WITH_GSSAPI"] = False
+        tc.variables["WITH_MBEDTLS"] = False
+        tc.variables["WITH_NACL"] = False
+        tc.variables["WITH_SYMBOL_VERSIONING"] = False
+        tc.variables["WITH_ZLIB"] = self.options.with_zlib
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.cache_variables["CMAKE_TRY_COMPILE_CONFIGURATION"] = str(self.settings.build_type)
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["ssh"]
+        if not self.options.shared:
+            self.cpp_info.defines.append("LIBSSH_STATIC=ON")
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs.extend(["ws2_32", "iphlpapi"])
+
+        self.cpp_info.set_property("cmake_file_name", "libssh")
+        self.cpp_info.set_property("cmake_target_name", "ssh")
+        self.cpp_info.set_property("pkg_config_name", "libssh")
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "libssh")
        self.cpp_info.set_property("cmake_target_name", "ssh::ssh")
        self.cpp_info.set_property("cmake_target_aliases", ["ssh"])
        self.cpp_info.set_property("pkg_config_name", "libssh")
```
The library exports both `ssh` and `ssh::ssh` as targets: https://git.libssh.org/projects/libssh.git/tree/src/CMakeLists.txt#n345

The rest are correct:
cmake config file: https://git.libssh.org/projects/libssh.git/tree/src/CMakeLists.txt#n387
.pc file: https://git.libssh.org/projects/libssh.git/tree/CMakeLists.txt#n102

You could also add `ssh-static` and `ssh::static` aliases if not self.options.shared: https://git.libssh.org/projects/libssh.git/tree/src/CMakeLists.txt#n407

Reply:
Including one or more of the above links as a reference in a comment would not hurt either.

Reply:
I added the missing targets, but I'm not sure which lines in particular you want comments on. Could you make a suggested change with what you are after and I'll add it in?

Reply:
Hi @valgur, I added some comments with links to the lines these definitions are in the libssh-0.10.6 tag as comments. Can you review this and let me know if it's what you had in mind?